test/modules/MetricsValidator/src/ValidateNumberOfMessagesSent.cs

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace MetricsValidator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Client;
+    using Microsoft.Azure.Devices.Edge.Agent.Diagnostics;
+
+    public class ValidateNumberOfMessagesSent : TestBase
+    {
+        readonly string endpoint = Guid.NewGuid().ToString();
+
+        public ValidateNumberOfMessagesSent(TestReporter testReporter, IMetricsScraper scraper, ModuleClient moduleClient)
+            : base(testReporter, scraper, moduleClient)
+        {
+        }
+
+        public override async Task Start(CancellationToken cancellationToken)
+        {
+            Console.WriteLine($"Starting {nameof(ValidateNumberOfMessagesSent)}");
+            await this.TestNumberSent(10, cancellationToken);
+            await this.TestNumberSent(100, cancellationToken);
```

Comment:
(Optional) Depending on the testing philosophy, but you could make these numbers random (with caps) and hope that every time the test run it expand the test coverage by a bit.

Reply:
I don't think that expands coverage, unless there's a really bizarre bug where it only counts in even numbers or something.

Reply:
Tempting to use random, but I'd prefer tests remain consistent and repeatable.